app/src/views/CountryNsOverviewContextAndStructure/NationalSocietyLocalUnits/LocalUnitsFormModal/LocalUnitsForm/index.tsx

```diff
@@ -509,7 +574,10 @@ function LocalUnitsForm(props: Props) {
 
     return (
         <div className={styles.localUnitsForm}>
-            {readOnly && isDefined(actionsContainerRef.current) && (
+            {!localUnitDetailsResponse?.is_locked
+                && readOnlyFromProps
+                && isNotDefined(localUnitDetailsResponse)
```

Comment:
Shouldn't this be `isDefined(localUnitDetailsResponse)` ?
Also, this check should come before `!localUnitDetailsResponse?.is_locked`